lib/src/utils/MuiPickersUtilsProvider.jsx

```diff
@@ -14,18 +17,10 @@ export default class MuiPickersUtilsProvider extends Component {
     moment: undefined,
   }
 
-  static childContextTypes = {
-    muiPickersDateUtils: PropTypes.object,
-  }
-
-  getChildContext() {
+  render() {
     const { utils: Utils, locale, moment } = this.props;
-    return {
-      muiPickersDateUtils: new Utils({ locale, moment }),
-    };
-  }
+    const utils = new Utils({ locale, moment });
```

Comment:
This can be put inside state and updated with `static getDerivedStateFromProps(nextProps, prevState)`. Having this inside render will cause cause all datepicker components to update. Because every time, utils will be a new instance and will force all consumers to update

Reply:
Actually this do the trick. Because of provider will be rerendered when locale has been changed - every consumer will rerender and displayed up-to-date localized text.